src/lib/get-locale.ts

```diff
@@ -0,0 +1,55 @@
+import type { CountryCode } from '@guardian/libs';
+import { getCookie, isString, storage } from '@guardian/libs';
+import type { Edition } from '../types';
+
+const KEY = 'GU_geo_country';
+const KEY_OVERRIDE = 'gu.geo.override';
+const COUNTRY_REGEX = /^[A-Z]{2}$/;
+
+// best guess that we have a valid code, without actually shipping the entire list
+const isValidCountryCode = (country: unknown) =>
+	isString(country) && COUNTRY_REGEX.test(country);
+
+// we'll cache any successful lookups so we only have to do this once
+let locale: CountryCode | undefined;
+
+const editionToGeolocationMap: Record<Edition, CountryCode> = {
+	UK: 'GB',
+	US: 'US',
+	AU: 'AU',
+};
+
+const editionToGeolocation = (editionKey: Edition): CountryCode =>
+	editionToGeolocationMap[editionKey];
+
+// just used for tests
+export const __resetCachedValue = (): void => (locale = void 0);
```

Comment:
It'd be nice to avoid defining and exporting a function used just for testing, but I can't think of a way around it in this instance!

A bit of a nitpick but I always find `void 0` a bit confusing, is this just acting as `undefined` here?

Reply:
Agreed on the unfortunate need to expose this function!

Also agreed on `void 0`, have updated